components/nimbus/src/cirrus.udl

```diff
@@ -0,0 +1,183 @@
+namespace cirrus {};
+
+dictionary AppContext {
+    string app_name;
+    string app_id;
+    string channel;
+    string? app_version = null;
+    string? app_build = null;
+    string? architecture = null;
+    string? device_manufacturer = null;
+    string? device_model = null;
+    string? locale = null;
+    string? os = null;
+    string? os_version = null;
+    string? android_sdk_version = null;
+    string? debug_tag = null;
+    i64? installation_date = null;
+    string? home_directory = null;
+    JsonObject? custom_targeting_attributes = null;
+};
+
+dictionary TargetingAttributes {
+    AppContext app_context;
+    boolean is_already_enrolled;
+    StringHashSet active_experiments;
+    string? language = null;
+    string? region = null;
+    i32? days_since_install = null;
+    i32? days_since_update = null;
+};
+
+dictionary ExperimentBranch {
+    string slug;
+    i32 ratio;
+};
+
+dictionary EnrollmentChangeEvent {
+    string experiment_slug;
+    string branch_slug;
+    string enrollment_id;
+    EnrollmentChangeEventType change;
+    string? reason = null;
+};
+
+enum EnrollmentChangeEventType {
+    "Enrollment",
+    "EnrollFailed",
+    "Disqualification",
+    "Unenrollment",
+    "UnenrollFailed",
+};
+
+dictionary ExperimentEnrollment {
+    string slug;
+    EnrollmentStatus status;
+};
+
+dictionary EnrolledFeatureConfig {
+    FeatureConfig feature;
+    string slug;
+    string feature_id;
+    string? branch = null;
+};
+
+dictionary FeatureConfig {
+    string feature_id;
+    JsonObject value;
+};
+
+[Enum]
+interface EnrollmentStatus {
+    Enrolled(Uuid enrollment_id, EnrolledReason reason, string branch);
+    NotEnrolled(NotEnrolledReason reason);
+    Disqualified(Uuid enrollment_id, DisqualifiedReason reason, string branch);
+    WasEnrolled(Uuid enrollment_id, string branch, u64 experiment_ended_at);
+    EnrollError(string reason);
+};
```

Comment:
This threw me off — I didn't realize that parameterized enums a) had a separate syntax, and b) required you to include all the parameters.

I like the syntax though, seems to handle it great and makes a lot of sense.